sdk-core/token_transfer.md

```diff
@@ -0,0 +1,53 @@
+## TokenTransfer
+
+```
+class TokenTransfer:
+    // Fields (properties):
+    token_identifier: ITokenIdentifier;
+    token_nonce: number;
+    amount_in_atomic_units: number;
+
+    constructor(tokenIdentifier: ITokenIdentifier, tokenNonce: number, amountInAtomicUnits: number);
+
+    isEgld(): boolean;
+
+    isFungible(): boolean;
+
+    // Named constructors:
+    static ofEgld(amountInAtomicUnits: number): TokenTransfer;
+    static ofFungible(tokenIdentifier: ITokenIdentifier, amountInAtomicUnits: number): TokenTransfer;
+    static ofNonFungible(tokenIdentifier: ITokenIdentifier, nonce: number): TokenTransfer;
+    static ofSemiFungible(tokenIdentifier: ITokenIdentifier, nonce: number, quantity: number): TokenTransfer;
+    static ofMetaESDT(tokenIdentifier: ITokenIdentifier, nonce: number, amountInAtomicUnits: number): TokenTransfer;
```

Comment:
alternatively, we can have a token type:
```
class Token:
    token_identifier: ITokenIdentifier;
    token_nonce: number;
    num_decimals: number;
    type: ITokenType;
    metadata: ITokenMetadata;
```
and then we can have a single named constructor:
```
static ofToken(token: Token): TokenTransfer;
```
and based on its type and number of decimals, it should be able to compute the transfer action

The disadvantage is that the dev should be careful about the type and have more items in the DTO than he needs (for example, when transferring EGLD), so I don't have a strong opinion. 

However, I don't think amountInAtomicUnits is the way to go. Why not use the `Amount` type instead?

Reply:
Thanks for the idea, will think about it. 

Perhaps we can have `ofNativeToken()` and `ofCustomToken()`?

At first, I wanted the functions to receive `Amount` as an interface having `toString()` or `toAtomicUnits()`. Will keep your idea on the table.